### DISCUSSION TITLE
Unable to install GNPy in windows 10

### DISCUSSION THREAD

**Discussion by NAIK131340** [Telecominfraproject/oopt-gnpy#467] [Q&A] [answered]
Hi Team,
I'm following the instructions given by url's below but unable to run all test cases after running pytest command.
https://gnpy.readthedocs.io/en/master/install.html
https://gnpy.readthedocs.io/en/master/install.html#installing-the-python-package
I'm using Windows 10 Machine, Python 3.11.0 and Anaconda

Please find the error details from below file after running pytest
[pytest.txt](https://github.com/Telecominfraproject/oopt-gnpy/files/11500384/pytest.txt)

Kindly please tell what is happening here

Kind regards
Naik

**Comment by felipebm** [comment]
Hi Naik,

I just did my installation on Win10, so I'll be happy to help.
what happens when you run python -c "import gnpy" ?

best,

*Felipe Beltran-Mejia*

*Optical Systems Developer*
www.padtec.com.br




Em qua., 17 de mai. de 2023 às 07:28, NAIK131340 ***@***.***>
escreveu:

**Comment by NAIK131340** [comment]
Hi Felipe,
Thanks for your response
I didn't got anything after running python -c "import gnpy"
but after pytest I'm getting 22 failed test cases. Attaching the error after running pytest
[pytest.txt](https://github.com/Telecominfraproject/oopt-gnpy/files/11500550/pytest.txt)
 please have a look
Thanks and regards
Naik

**Comment by jktjkt** [comment, marked as answer]
The correct installation command is:
```
pip install --editable .
```
...whereas as per the attached log, you actually used this:
```
pip install --editable
```
That resulted in an error message because the trailing period is missing. The period/dot refers to the [current working directory](https://en.wikipedia.org/wiki/Working_directory), and it tells `pip` where to install from.

**Comment by NAIK131340** [comment]
Thank you for the help but after successfully installation.
 After checking python -c "import gnpy"
checking pytest error is coming attaching the same
[error.txt](https://github.com/Telecominfraproject/oopt-gnpy/files/11500942/error.txt)
Kindly please tell what is happening here

Kind regards
Naik

**Reply by jktjkt** [reply]
From the attached log it seems that you've cloned the repository a few times, so that now you have a copy of the repo within another copy of the other repo:
```
    raise ImportPathMismatchError(module_name, module_file, path)
E   _pytest.pathlib.ImportPathMismatchError: ('tests.conftest', 'C:\\Users\\naik1\\oopt-gnpy\\tests\\conftest.py', WindowsPath
('C:/Users/naik1/oopt-gnpy/gnpy/example-data/oopt-gnpy/oopt-gnpy/tests/conftest.py'))
```
I suggest either removing everything (including the Python virtual environment) and starting once again from scratch. Maybe just removing `C:/Users/naik1/oopt-gnpy/gnpy/example-data/oopt-gnpy` might do the trick as well.

**Comment by NAIK131340** [comment]
Thank you so much for the help.
Same error is occuring after removing  python virtual environment and started from the scratch also

**Reply by jktjkt** [reply]
Please post a complete console log of all the commands that you've used, starting from scratch, as well as the output of `pip list`.

**Reply by NAIK131340** [reply]
Its working Now Thank you so much Jan. Thanks for the help
